framework/src/main/java/org/tron/core/services/http/TriggerConstantContractServlet.java

```diff
@@ -65,9 +71,20 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
       validateParameter(contract);
       JsonFormat.merge(contract, build, visible);
       JSONObject jsonObject = JSONObject.parseObject(contract);
-      String selector = jsonObject.getString(functionSelector);
-      String parameter = jsonObject.getString("parameter");
-      String data = Util.parseMethod(selector, parameter);
+
+      boolean isDataSet = jsonObject.containsKey("data")
+          && !StringUtil.isNullOrEmpty(jsonObject.getString("data"));
+
+      String data;
+
+      if (isDataSet) {
```

Comment:
Recommend to use `isFuntionSelectorSet` here in order to prevent some projects already use `data` in their logic. So, you may need to reverse the content in `if` and `else`.
Also, please propose a TIP and we will consider this in the next release.